interface.h

```diff
@@ -64,150 +67,853 @@
+ *        to "shape").
+ * 
+ * @param mode 0 default behaviour
+ *             1 activates alternate modes with any non-zero value treated as 1
+ */
 void set_jf_mode(uint8_t mode);
+
+/**
+ * @brief Set the value of the indicated CV gate output on a connected Just
+ *        Friends via I2C.
+ * 
+ * @param output Just Friends CV gate output number. A valid value is between 0
+ *        and MAX_JF_VOICE_COUNT - 1
+ * @param on 0 CV gate output is low,
+ *           1 CV gate output is high
+ */
 void set_jf_gate(uint8_t output, uint8_t on);
 
+/**
+ * @brief Set the envelope mode of the indicated output on a connected TELEXo
+ *        module via I2C.
+ * 
+ * @param output TELEXo CV output number. A valid value is between 0 and
+ *        MAX_TXO_VOICE_COUNT - 1
+ * @param mode TELEXo CV output envelope mode
+ *             0 sets output to oscillator mode and initializes the output to
+ *             waveform 0 (sine)
+ *             1 sets output to envelope mode
+ * @see set_txo_waveform()
+ */
 void set_txo_mode(uint8_t output, uint8_t mode);
+
+/**
+ * @brief Set the CV value of the indicated output on a connected TELEXo module
+ *        via I2C.
+ * 
+ * @param output TELEXo CV output number. A valid value is between 0 and
+ *        MAX_TXO_VOICE_COUNT - 1
+ * @param value CV value represented as a signed integer with a range of
+ *        -16,384 (-10v) and 16,383 (+10v)
+ */
 void set_txo_cv(uint8_t output, int16_t value);
+
+/**
+ * @brief Set the CV gate value of the indicated output on a connected TELEXo
+ *        module via I2C.
+ * 
+ * @param output TELEXo CV gate output number. A valid value is between 0 and
+ *        MAX_TXO_VOICE_COUNT - 1
+ * @param on 0 CV gate output is low,
+ *           1 CV gate output is high
+ */
 void set_txo_gate(uint8_t output, uint8_t on);
 
+/**
+ * @brief Set the envelope attack value of the indicated output on a connected
+ *        TELEXo module via I2C.
+ * 
+ * @param output TELEXo output number. A valid value is between 0 and
+ *        MAX_TXO_VOICE_COUNT - 1
+ * @param attack Output envelope attack rate in milliseconds. A valid value is
+ *        between 1 and n
+ */
 void set_txo_attack(uint8_t output, uint16_t attack);
+
+/**
+ * @brief Set the envelope decay value of the indicated output on a connected
+ *        TELEXo module via I2C.
+ * 
+ * @param output TELEXo output number. A valid value is between 0 and
+ *        MAX_TXO_VOICE_COUNT - 1
+ * @param decay Output envelope decay rate in milliseconds. A valid value is
+ *        between 1 and n
+ */
 void set_txo_decay(uint8_t output, uint16_t decay);
+
+/**
+ * @brief Set the waveform of the indicated CV output in oscillator mode on a connected TELEXo module
+ *        via I2C.
+ * 
+ * @param output TELEXo output number. A valid value is between 0 and
+ *        MAX_TXO_VOICE_COUNT - 1
+ * @param waveform Desired output waveform. A valid value is between 0 and 325
```

Comment:
the range is:

for teensy 3.2: `0-4999`; values translate to sine (0), triangle (1000), saw (2000), pulse (3000), or noise (4000); oscillator shape between values is a blend of the pure waveforms

for teensy 3.6: `0-4500`. There are 45 different waveforms, values translate to sine (0), triangle (100), saw (200), pulse (300) all the way to random/noise (4500); oscillator shape between values is a blend of the pure waveforms